app/routes.py

```diff
@@ -31,22 +81,32 @@ def add_user():
 def manage_user(user_id):
     if request.method == "PUT":
         data = request.get_json(force=True)
-        username = data["username"]
-        password = data["password"]
+        username = data.get("username")
+        if username == None:
+            return jsonify({"msg":"username is required"})#3
+        password = data.get("password")
```

Comment:
you can merge `if not username or not password` so you will have only one check.